backend/src/test/java/ai/verta/modeldb/GlobalSharingTest.java

```diff
@@ -0,0 +1,451 @@
+  }
+
+  private static final Collection<Object[]> requestData =
+      Arrays.asList(
+          new Object[][] {
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_WRITE),
+              true,
+              true
+            },
+            {ResourceVisibility.ORG_DEFAULT, CollaboratorPermissions.newBuilder(), true, false},
+            {
+              ResourceVisibility.ORG_CUSTOM,
+              CollaboratorPermissions.newBuilder().setCollaboratorType(CollaboratorType.READ_ONLY),
+              true,
+              false
+            },
+            {ResourceVisibility.PRIVATE, CollaboratorPermissions.newBuilder(), false, false}
+          });
+
+  @Test
+  public void createProjectWithGlobalSharingOrganization() {
+    LOGGER.info("Global organization Project test start................................");
+
+    if (app.getAuthServerHost() == null || app.getAuthServerPort() == null) {
+      Assert.assertTrue(true);
+      return;
+    }
+
+    String orgName = "Org-test-verta";
+    SetOrganization setOrganization =
+        SetOrganization.newBuilder()
+            .setOrganization(
+                Organization.newBuilder()
+                    .setName(orgName)
+                    .setDescription("This is the verta test organization")
+                    .build())
+            .build();
+    SetOrganization.Response orgResponse =
+        organizationServiceBlockingStub.setOrganization(setOrganization);
+    Organization organization = orgResponse.getOrganization();
+    assertEquals(
+        "Organization name not matched with expected organization name",
+        orgName,
+        organization.getName());
+
+    String orgRoleName = "O_" + organization.getId() + "_GLOBAL_SHARING";
+    GetRoleByName getRoleByName =
+        GetRoleByName.newBuilder()
+            .setName(orgRoleName)
+            .setScope(RoleScope.newBuilder().setOrgId(organization.getId()).build())
+            .build();
+    GetRoleByName.Response getRoleByNameResponse =
+        roleServiceBlockingStub.getRoleByName(getRoleByName);
+    assertEquals(
+        "Expected role name not found in DB",
+        orgRoleName,
+        getRoleByNameResponse.getRole().getName());
+    organizationServiceBlockingStub.addUser(
+        AddUser.newBuilder()
+            .setOrgId(organization.getId())
+            .setShareWith(authClientInterceptor.getClient2Email())
+            .build());
+
+    String orgResourceId = null;
+    Repository repository = null;
+    try {
+      for (Object[] data : requestData) {
+        ResourceVisibility resourceVisibility = (ResourceVisibility) data[0];
+        CollaboratorPermissions.Builder customPermission =
+            (CollaboratorPermissions.Builder) data[1];
+        boolean isReadAllowed = (boolean) data[2];
+        boolean isWriteAllowed = (boolean) data[3];
+        if (orgResourceId == null) {
```

Comment:
This will change as part of the loop and it's pretty confusing to me how things relate in this test because of this.

Reply:
it checks if it should create or update resource. If orgResourceId is null it means that it is not created

Reply:
@evgeniilezhnin `orgResourceId` is not null then as per code logic it has failed some where and not clear the created resources right? and you ignore all other next text scenario from for loop. Also if `orgResourceId` is nut null then you are not throwing error and mark test case fail.

Reply:
The logic is: there is a loop on different TestParameters. On a first iteration it creates a resource. On next iterations it updates an existing resource with current TestParameters. When the test ends there is a resource deletion.